### DISCUSSION TITLE
Grand Org Crescendo and volume pedals

### DISCUSSION THREAD

**Discussion by larspalo** [GrandOrgue/grandorgue#415] [General]
Originally posted 2021-08-01 11:02:40.596000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>Hi.
I builded already organ instrument. When I started built it I thought to use Grand Orgue. Very usefull was movies by Lars Palo. I can play for it now, but I can&#039;t connect volume pedal
and crescendo pedal to GO. I try to do it this same way as is a pedal key connect with commend - Panel- Crescendo pedal or coupler manuals and Volume. When I chose right mouse 
on picture of the pedal then apears card. In this card I use - listen for Event and mooving the pedal then is described in channell 6. It is right first used channel in my organ.  Then I used 
Detect complex Midi setup mowing pedal on min, then on max positions. In line Event I use all option, but no efect when pushing OK, because opening this card again, it is empty. I&#039;ve tried 
to joint this devices to place where the pedal keys is joined, then I can change voices from C to g&#039; mooving angle of this pedal. Then, I think, my device works properly, I think. In my organ I have one pedal Crescendo,
it is woking in channel 6, and two pedals volume joined to channels 7 and 8. all this devices has 32 levels. Below I show all my working device.

Klawiatura 3 Keystation 61 4 ( from bottom )
9xNote
Channel 14 
transpose 0
Key 36   96

Klawiatura 2 Keystation 61 2 ( from bottom )
9xNote
Channel 1 
transpose 0
Key 36   96

Klawiatura 1 Keystation 61 0 ( from bottom )
9xNote
Channel 5
transpose 0
Key 36   96

Pedal MidiIN4(MIDI4x4)9
9xNote
Channel 5  
Key 36 - 67

Wybór głosów na Panelu (pokrywie)
General ( card )
Channel 5
Przyciski    1 - 8   =&gt; 92 - 93 - 94 - 95 - 96 - 33 - 34 - 35
               9 - 16  =&gt; 84 - 85 - 86 - 87 - 88 - 89 - 90 - 91
             17 - 24  =&gt; 76 - 77 - 78 - 79 - 80 - 81 - 82 - 83
             25 - 30  =&gt; 68 - 69 - 70 - 71 - 72 - 73

I can to share all document which I created. You can do a copy of my organ.

ps.
to Lars Palo. I&#039;ve sent You mail 20210710 on burea.fxxxxxg@svxxxxxyrkan.se. You Can see my organ.

**Comment by larspalo** [comment]
Originally posted 2021-08-02 06:49:07.625000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>I suspect your pedals are not sending any midi events to your computer. Pedals normally produce midi &quot;Control Change&quot; events or&quot; NRPN&quot; events.

If you are using Windows then use Midi-Ox to check what (if any) midi signals your pedals are sending to the computer. If Linux then use an appropriate alternative.

csw900

**Reply by larspalo** [reply]
Originally posted 2021-08-02 18:20:02.408000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>I&#039;ve checked this signal, when I using card Midi settings answer is properly (Listen for event and Detect complex..), and even I tried to change leads from pedals key then I can play pedal souds moving pedals crescendo or shatters.  I use Windows 10.  How I can use Midi-Ox. Can You explain it. May be there is solve the problem.

**Comment by larspalo** [comment]
Originally posted 2021-08-02 18:53:34.918000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Download Midi-Ox (use google to find the web site). Install it and run it (do NOT run GO at the same time). Make sure it is connected to your organ console midi output - you should know it name because it will be the same output that GO uses.

When it is running it will show you exactly what midi signals your computer is receiving. Test it by playing some notes. Then move one of the pedals - you will see any midi that the pedals are producing. If you see nothing then the pedals are not sending midi and this is why they do not work.

csw900

**Comment by larspalo** [comment]
Originally posted 2021-08-03 18:49:43.444000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>I&#039;ve running Midi-Ox. When choice in carrd Options, Midi Devices all my using devices then all of them working, pedals too. A little problem is, the devices name is changing during my working, then I decided using all. For me is not bad because I only serching not working devices.
Thank You for suggestion, but I&#039;m still on the beginnig.

**Comment by larspalo** [comment]
Originally posted 2021-08-04 07:10:22.889000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Glad you have midi-ox working - you will learn a lot from it.

Regarding the interface name changes: The names are fixed and cannot be altered - however if you have more than one interface with the same name GO will add a number. For example if you have two interfaces named &quot;MIDI interface&quot; then GO will number them &quot;MIDI interface 1&quot;and &quot;MIDI interface 2&quot;.  This does NOT help to identify the interfaces and can cause much confusion because the numbers are not fixed, they depend upon the order the interfaces were found when the computer was booted and this is can vary unpredictably. Thus &quot;MIDI interface 1&quot; can be the swell manual one day and next time you boot it could be the pedals.

I always recommend getting every midi to USB adapter cable from a different makers so that their names are different and this cures the problem.

csw900

**Comment by larspalo** [comment]
Originally posted 2021-08-04 08:43:22.754000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>Jerzy, what OS do you use with GO? I know a solution for linux, but don&#039;t know any for other systems.

**Comment by larspalo** [comment]
Originally posted 2021-08-04 13:49:44.282000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>Thanks CSW900 I will think about the name of my MIDI interface. Will this solution fix my problem? I use Windows 10, but I have already installed Linux Ubuntu 20.04 on my computer together. I would start Go on this system, but I&#039;m not good in this system. I have driver for monitor Asus MB16AMT  for Ubuntu, but I don&#039;t know how to install it. And the program GO I have, but I don&#039;t know how I can install it. I looked about it, but no one is explaining Ubuntu 20.04.

**Reply by larspalo** [reply]
Originally posted 2021-08-04 20:02:09.125000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>&gt; Will this solution fix my problem? I

Which solution? Using usb-midi cables of different vendors - yes, if OS gives them different names. Some vendors may use  the same chip for their cables, so OS cann&#039;t differentiate them.

If you are talking about linux: yest. there is some solution under Linux: matching with the USB ports instead the device names. It requires some configuration files and scripts.

&gt; I have driver for monitor Asus MB16AMT for Ubuntu, but I don&#039;t know how to install it

First try to use it without any additional driver. Usually the only problem with touchscreens on Linux is if you have two identical touchscreens.

&gt;  And the program GO I have, but I don&#039;t know how I can install it. I looked about it, but no one is explaining Ubuntu 20.04.

https://github.com/oleg68/GrandOrgue/blob/master/INSTALL.md Read &quot;Installation on debian-based systems (debian 9+, ubuntu 18+, mint, raspbian)&quot;

**Reply by larspalo** [reply]
Originally posted 2021-08-05 13:29:28.370000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Hi Oleg

I am interested in your alternative method of identifying midi input ports - if it works for Linux it will also work for Windows. If it involves the DeviceID&#039;s of the midi ports (these vary from 0 to 1 less than the number of midi ports) then I suspect that these are also numbered in the order that the OS finds them when booting rather than being fixed. Consequently they will be no more reliable than the device names for uniquely identifying midi sources. As far as I know USB ports do not have any unique identifying feature.

I could be wrong - have you tested and proved your method?

csw900

**Reply by larspalo** [reply]
Originally posted 2021-08-05 18:50:00.486000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>Hi, csw900

&gt; As far as I know USB ports do not have any unique identifying feature. 
&gt; have you tested and proved your method?

Yes, of cause. I use it in my organ.

Linux uses USB paths like `/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3.1/1-3.1.4/1-3.1.4:1.0` and `/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3.2/1-3.2:1.2` for identifying usb ports. So I wrote an udev rule that names audio and midi interfaces based on their paths:

~~~
SUBSYSTEM!=&quot;sound&quot;, GOTO=&quot;my_audio_end&quot;
ACTION!=&quot;add&quot;, GOTO=&quot;my_audio_end&quot;
DEVPATH==&quot;/devices/pci0000:00/0000:00:01.0/0000:01:00.1/sound/card?&quot;, ATTR{id}=&quot;NVidia1650&quot;
DEVPATH==&quot;/devices/pci0000:00/0000:00:01.1/0000:02:00.1/sound/card?&quot;, ATTR{id}=&quot;NVidia640&quot;
DEVPATH==&quot;/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3.1/1-3.1.3/1-3.1.3:1.0/sound/card?&quot;, ATTR{id}=&quot;U610_2&quot;
DEVPATH==&quot;/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3.1/1-3.1.4/1-3.1.4:1.0/sound/card?&quot;, ATTR{id}=&quot;U610_3&quot;
DEVPATH==&quot;/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3.2/1-3.2:1.2/sound/card?&quot;, ATTR{id}=&quot;Webcam_Organ&quot;
DEVPATH==&quot;/devices/pci0000:00/0000:00:14.0/usb1/1-4/1-4.4/1-4.4.4/1-4.4.4.4/1-4.4.4.4:1.2/sound/card?&quot;, ATTR{id}=&quot;Webcam_Mon&quot;
LABEL=&quot;my_audio_end&quot;
~~~

Then I have ALSA card names U610_2 for the second manual and U610_3 for the third manual. I use a shell script that connects midi ports of their ALSA names to GrandOrgue.

I&#039;m an expert neither in Windows nor in OSx, so I don&#039;t have any ideas how to make such tweak with these OSes.

**Reply by larspalo** [reply]
Originally posted 2021-08-06 13:31:27.600000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Hi Oleg

VERY interesting. Which linux do you use?

I have just had a quick look at a Ubuntu 20 shell (Running under Windows) and it does not have a &quot;/devices&quot; folder at all, but does have &quot;/dev&quot;. 

Looking in /dev it does not list usb ports. It mainly lists loads of unused tty ports.
Looking into /mnt there are no usb ports listed. Just the disk drives. Maybe the windows ubuntu has been cut down too much.

Looking at a proper full version of Ubuntu the usb ports are listed under /dev/bus/usb/001/001 and /dev/bus/usb/001/002 etc. It also has the udev feature so your method should work on Ubuntu Linux but would need its own version of the script files.

On Windows I have never seen usb ports listed at all unless they are connected to something and even then it depends on what they are connected to - such as a disk drive. However the usb and all other devices can be listed by
pnputil  /enum-devices
This lists everything windows knows about devices including those no longer connected. It is pretty useless. Also windows has no equivalent to udev.

Thus, unfortunately, I do not think your method can be made to work for windows.

csw900

**Reply by larspalo** [reply]
Originally posted 2021-08-06 16:08:24 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>&gt; I have just had a quick look at a Ubuntu 20 shell (Running under Windows) and it does not have a &quot;/devices&quot; folder at all, but does have &quot;/dev&quot;.

Udev rules are written relative the /sys folder. So the full path is ``/sys/devices/pci0000:00/0000:00:14.0/usb1/1-3/1-3.1/1-3.1.3/1-3.1.3:1.0``

&gt;  Which linux do you use?

I use Fedora 34 but this rule can run in all systemd/udev based linux distributions, including Ubuntu. `/sys` is a `sysfs` virtual filesystem that contains information about all devices in the system.

&gt;  I do not think your method can be made to work for windows.

I don&#039;t know, does Windows emulate `sysfs` in WSL?

**Reply by larspalo** [reply]
Originally posted 2021-08-07 10:04:27.526000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Oleg.
      does Windows emulate sysfs in WSL?
      
      I do not think it does. I tried
      sudo mount -t sysfs /sys
      the result was 
      /sys: cant find in /etc/fstab
      
      In fact /etc/fstab does not exist.
      
      There is a /sys/devices folder but it contains no usb ports.
      
      The Windows Ubuntu appears to have very limited interaction
   capabilities with Windows so even if your udev method could be
    made to work I do not think it would be of any benefit to the
    Windows GO.
      
      csw900

**Comment by larspalo** [comment]
Originally posted 2021-08-04 14:23:35.181000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Your ASUS touch screen monitor should work without installing any driver. I use a DELL touch screen monitor myself.  Just plug it in and it will work. It should have a USB output which provides mouse signals to the computer when you touch the screen. I suggest you get GO working properly on your windows system before playing about with Ubuntu.

csw900

**Reply by larspalo** [reply]
Originally posted 2021-08-04 15:05:52.737000 by steve181 in GrandOrgue discussion forum on SourceForge<br><br>I use an Acer Touchscreen monitor with GO on Win 10. Just as CSW said: 1 
plug for video, 1 USB for control. It worked the first time. (I realize 
its not Ubantu.)

Steve

On 8/4/2021 10:23 AM, csw900 wrote:
&gt;
&gt; Your ASUS touch screen monitor should work without installing any 
&gt; driver. I use a DELL touch screen monitor myself. Just plug it in and 
&gt; it will work. It should have a USB output which provides mouse signals 
&gt; to the computer when you touch the screen. I suggest you get GO 
&gt; working properly on your windows system before playing about with Ubuntu.
&gt;
&gt; csw900
&gt;
&gt; ------------------------------------------------------------------------
&gt;
&gt; Grand Org Crescendo and volume pedals 
&gt; &lt;https://sourceforge.net/p/ourorgan/discussion/962124/thread/63a5c9a5a3/?limit=25#63d1&gt;
&gt;
&gt; ------------------------------------------------------------------------
&gt;
&gt; Sent from sourceforge.net because you indicated interest in 
&gt; https://sourceforge.net/p/ourorgan/discussion/962124/ 
&gt; &lt;https://sourceforge.net/p/ourorgan/discussion/962124/&gt;
&gt;
&gt; To unsubscribe from further messages, please visit 
&gt; https://sourceforge.net/auth/subscriptions/ 
&gt; &lt;https://sourceforge.net/auth/subscriptions/&gt;
&gt;
<br><br>Attachments in original post:<br>https://sourceforge.net/p/ourorgan/discussion/962124/thread/63a5c9a5a3/63d1/7987/attachment/alternate<br>

**Comment by larspalo** [comment]
Originally posted 2021-08-06 13:39:24.364000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>I have  four canals for data. It&#039;s number (by hardwer described) is 5, 6, 7 and 8. 

They are two devices and each of them has two canals 5, 6 and 7, 8. Every cannals have 128 bits.
This two devices are connected to one device midiface 4x4 -manufacturer Miditech. 

Then I think there no conflicts are, because there is only one device. I checked exactly  only canal 5 i partly ( two bits of signal ) from canal number 6 from first device.  

The second device I&#039;ve checket only by Midi-Ox. Idon&#039;t know is it enough?  Of course first device
 was checked by Midi-Ox too. 

May be I&#039;ve maked mistakes when I try connect pedals crescendo or volume to GO. Can You explain it for me in detail?
For me very important information is how many levels ( steps) can GO read in crescendo and volume pedal. Plese tell me this informations. 
It is necessary to make a new program for controlling organ processor. 

 
 Asus Scren MB16 AMT under Ubuntu 20.04 without driver.
Actually touch is working, but there aren&#039;t any pictures or icons on it (only dark screen).

I maked touch on Asus monitor and saw result on computer screen, because Asus monitor is dark. 
I think driver - displaylink-driver-5.3.1.34.run is necessary to start a screen, but I don&#039;t know how to install it.
But You&#039;ve described how to install GO in Linux. For me it is a Jungle. Then I must to stay with Windows, 
or I will get a describe how to install it from one file which I have - grandorgue_0.3.1.2334-1+155.1_amd64.deb. 

May be later I will found a lot of time to study Linux and will do it myself.

**Reply by larspalo** [reply]
Originally posted 2021-08-06 16:17:55.068000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>&gt; or I will get a describe how to install it from one file which I have - grandorgue_0.3.1.2334-1+155.1_amd64.deb.

This file is not sufficient for ubuntu. You should download also ``grandorgue-resources_0.3.1.2334-1+155.1_all.deb`` and ``grandorgue-demo_0.3.1.2334-1+155.1_all.deb`` and install all tese files with one shell command

```
apt-get install ./grandorgue_0.3.1.2334-1+155.1_amd64.deb ./grandorgue-resources_0.3.1.2334-1+155.1_all.deb grandorgue-demo_0.3.1.2334-1+155.1_all.deb
```

**Reply by larspalo** [reply]
Originally posted 2021-08-06 16:17:57.146000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>&gt; Asus Scren MB16 AMT under Ubuntu 20.04 without driver.
Actually touch is working, but there aren&#039;t any pictures or icons on it (only dark screen).

It seems the problems are not with your monitor but with your videocard. Does any other monitor work on this system in Linux?

**Comment by larspalo** [comment]
Originally posted 2021-08-06 14:10:21.586000 by [csw900](https://github.com/midimusic) in GrandOrgue discussion forum on SourceForge<br><br>Regarding the ASUS screen - you appear to have two monitors connected to Windows. In order for them both to work you must use the Windows Control Panel to set up windows so that it knows about the second monitor. Both will then work.

The Midiface 4X4 has 4 midi inputs and converts them to USB for connection to the computer. GO will see this as one device and you will see its name when you set up the midi inputs to GO. I assume you are using this to connect 3 manuals + pedal board. I assume these are the channels 5,6,7 and 8. I would expect this to work OK with GO without problems.

You have not said how you are connecting the swell and crescendo pedals but I assume you are using two midi to usb adapter cables which GO will see and connect to. I would also expect this to work OK with GO. The swell and cresendo pedals will each have 128 levels as defined by midi Control or NRPN events. You have not said what midi is being produced by these pedals. Midi Ox will tell you and you should tell us what MidiOx sees. It should make no difference to GO what midi channels are used here.

csw900

**Comment by larspalo** [comment]
Originally posted 2021-08-07 09:12:06.824000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>Hi Oleg.  You know, I found in controll panel card with monitors, but not one monitor. Then I think monitor Mb16AMT is detected, because it shows data from this monitor. It shows name of this monitor &quot;unknown&quot; Computer monitor is&#039;nt present on this card and I can&#039;t find its data anywhere, But it is only working. I havn&#039;t any other monitor, But in windows this monitor is working correctly,then videocards I think is good.
Hi CSW 900. I must hav more time to analise Your suggestions because of family and the days off.

**Comment by larspalo** [comment]
Originally posted 2021-08-08 12:30:31.520000 by [Oleg Samarin](https://github.com/oleg68) in GrandOrgue discussion forum on SourceForge<br><br>Hi, Jerzy!

As I understand, you have three issues:
1. Making volume pedals working
2. Installing GO on Ubuntu
3. Making your ASUS screen working on Ubuntu (not related to GO).

Let start with 1.
1. As I uderstand, you can make volume working working with ``Listen for a complex midi event``, but this settings disappeared when you start GO next time. The solution is File -&gt; Save after you has set it up.
2, I&#039;ve posted you a short instruction how to do it. Have you tried it?
3. Please describe me your hardware: is it a notebook or a desktop computer, which cable the ASUS monitor connected to. If there is a desktop, is there another screen where you can see what happens?

**Comment by larspalo** [comment]
Originally posted 2021-08-14 18:07:07.696000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>Hi Oleg, and csw900
Of course I have this three  issues. 
1. It didn&#039;t work at the beginning. It can&#039;t save by close tabsWhen open it next time in this same sesion it is empty.
Now I will go to holidays for two weeks, Then I can&#039;t  do anything. After two weeks I will do Yours suggestions., and answer about it.

**Comment by larspalo** [comment]
Originally posted 2021-08-14 18:07:11.295000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>Hi Oleg, and csw900
Of course I have this three  issues. 
1. It didn&#039;t work at the beginning. It can&#039;t save by close tabsWhen open it next time in this same sesion it is empty.
Now I will go to holidays for two weeks, Then I can&#039;t  do anything. After two weeks I will do Yours suggestions., and answer about it.

**Comment by larspalo** [comment]
Originally posted 2021-09-03 20:15:14.690000 by jerole in GrandOrgue discussion forum on SourceForge<br><br>Hi.  I thougt that may be I&#039;ve  made wrong steps during connection creshendo i volume pedal to GO. Because I&#039;ve never find information about it. Now I describe my  steps. I&#039;ve run Pitea school organ. In main bar I chose -Panel-Coupler manuals and volume or Crescendo pedal. I&#039;ve got chart where are icons of pedals. I presset on first  by right mouse key, and I&#039;ve got a chart  Midi settings for Enclosure Master or crescendo pedal. Then one more time I&#039;ve pressed on icon of pedal and apears a chart. Then I&#039;ve press listen for Event and moving a real pedal. Then some lines are changing. it apears number of canal, It is right chose, because the program of devices has this canal described in software. Then I&#039;ve preset Detect compleks Midi setup i moving the pedal in Max i Min position. All data change during this operation. Then press OK. Now I open this chart again, and it is empty. Where I&#039;ve made mistakes.

**Comment by jerole8231** [comment]
Hi. I'm wrote theme Grand Org Crescendo and Volume pedals #415. I've solve it myself this way
1. Monitor Asus to computer. Ive took them togheter to the service then they joined it to komputer by the USB2.0 and used controller from Microsoft not a orginal from Asus. Then it started works. it is stuped but works.
2. Midi device for volume shoe (pedal). I've taken a risk and bought a device wchich use a potentiometr translate to signal MIDI. Now it works excellent. Of course my old solution manages that device. I move shoe pedal
that is moving a step engine which produces impuses signal then I cout it by MIDI signal.
That all organ works properly.And one of Midi devices were crshed. After repair I can't to join pedals shoe to volume of keyboards or pedals. Some connections are working from last version (before reparing) but they moving wrong devices.
I've tried a movies by Lars Palo but I don't understand which devie he touch. Chart initial midi cofiguration and column Midi event stay always No. Can anybody to explain me step by step to solution this problem?
